From 7946586c33503bc383403faec48ffcea39e365ac Mon Sep 17 00:00:00 2001
From: Raphael Kim <raphk@google.com>
Date: Wed, 16 Oct 2024 14:42:21 -0700
Subject: [PATCH] [CDM] Check for system calling UID when backing up and
 restoring data

Bug: 370962373
Test: Manual test on POC
Flag: EXEMPT bugfix
(cherry picked from commit 692cddfb32abae6c77b00c4850fd36b7eaaf8c70)
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:191638ababfc5b03d63264b8932c5903f18543ba)
Merged-In: I307940b7194dd17eacbbfb8be5c92ac21e688c65
Change-Id: I307940b7194dd17eacbbfb8be5c92ac21e688c65
---
 .../server/companion/CompanionDeviceManagerService.java     | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/services/companion/java/com/android/server/companion/CompanionDeviceManagerService.java b/services/companion/java/com/android/server/companion/CompanionDeviceManagerService.java
index 57eadf65f2a0..de060bd78e28 100644
--- a/services/companion/java/com/android/server/companion/CompanionDeviceManagerService.java
+++ b/services/companion/java/com/android/server/companion/CompanionDeviceManagerService.java
@@ -724,11 +724,17 @@ public class CompanionDeviceManagerService extends SystemService {
 
         @Override
         public byte[] getBackupPayload(int userId) {
+            if (getCallingUid() != SYSTEM_UID) {
+                throw new SecurityException("Caller must be system");
+            }
             return mBackupRestoreProcessor.getBackupPayload(userId);
         }
 
         @Override
         public void applyRestoredPayload(byte[] payload, int userId) {
+            if (getCallingUid() != SYSTEM_UID) {
+                throw new SecurityException("Caller must be system");
+            }
             mBackupRestoreProcessor.applyRestoredPayload(payload, userId);
         }
 
-- 
2.47.1.613.gc27f4b7a9f-goog

